activerecord/test/cases/scoping/relation_scoping_test.rb

```diff
@@ -228,6 +228,14 @@ def test_current_scope_does_not_pollute_sibling_subclasses
       assert SpecialComment.all.any?
     end
   end
+
+  def test_circular_joins_with_current_scope_does_not_crash
+    assert_nothing_raised do
```

Comment:
Can you change the test to demonstrate the expected behavior instead of just `assert_nothing_raised`?